sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/SystemEventNames.java

```diff
@@ -435,136 +472,150 @@
-        put(COMMUNICATION_CHAT_THREAD_WITH_USER_DELETED, AcsChatThreadWithUserDeletedEventData.class);
-        put(COMMUNICATION_SMS_DELIVERY_REPORT_RECEIVED, AcsSmsDeliveryReportReceivedEventData.class);
-        put(COMMUNICATION_SMS_RECEIVED, AcsSmsReceivedEventData.class);
-
-        // Web events
-        put(WEB_APP_UPDATED, WebAppUpdatedEventData.class);
-        put(WEB_BACKUP_OPERATION_STARTED, WebBackupOperationStartedEventData.class);
-        put(WEB_BACKUP_OPERATION_COMPLETED, WebBackupOperationCompletedEventData.class);
-        put(WEB_BACKUP_OPERATION_FAILED, WebBackupOperationFailedEventData.class);
-        put(WEB_RESTORE_OPERATION_STARTED, WebRestoreOperationStartedEventData.class);
-        put(WEB_RESTORE_OPERATION_COMPLETED, WebRestoreOperationCompletedEventData.class);
-        put(WEB_RESTORE_OPERATION_FAILED, WebRestoreOperationFailedEventData.class);
-        put(WEB_SLOT_SWAP_STARTED, WebSlotSwapStartedEventData.class);
-        put(WEB_SLOT_SWAP_COMPLETED, WebSlotSwapCompletedEventData.class);
-        put(WEB_SLOT_SWAP_FAILED, WebSlotSwapFailedEventData.class);
-        put(WEB_SLOT_SWAP_WITH_PREVIEW_STARTED, WebSlotSwapWithPreviewStartedEventData.class);
-        put(WEB_SLOT_SWAP_WITH_PREVIEW_CANCELLED, WebSlotSwapWithPreviewCancelledEventData.class);
-        put(WEB_APP_SERVICE_PLAN_UPDATED, WebAppServicePlanUpdatedEventData.class);
-
-        // Machine Learning events
-        put(MACHINE_LEARNING_DATASET_DRIFT_DETECTED, MachineLearningServicesDatasetDriftDetectedEventData.class);
-        put(MACHINE_LEARNING_MODEL_DEPLOYED, MachineLearningServicesModelDeployedEventData.class);
-        put(MACHINE_LEARNING_MODEL_REGISTERED, MachineLearningServicesModelRegisteredEventData.class);
-        put(MACHINE_LEARNING_RUN_COMPLETED, MachineLearningServicesRunCompletedEventData.class);
-        put(MACHINE_LEARNING_RUN_STATUS_CHANGED, MachineLearningServicesRunStatusChangedEventData.class);
-
-        // Key Vault events
-        put(KEY_VAULT_CERTIFICATE_NEW_VERSION_CREATED, KeyVaultCertificateNewVersionCreatedEventData.class);
-        put(KEY_VAULT_CERTIFICATE_NEAR_EXPIRY, KeyVaultCertificateNearExpiryEventData.class);
-        put(KEY_VAULT_CERTIFICATE_EXPIRED, KeyVaultCertificateExpiredEventData.class);
-        put(KEY_VAULT_KEY_NEW_VERSION_CREATED, KeyVaultKeyNewVersionCreatedEventData.class);
-        put(KEY_VAULT_KEY_NEAR_EXPIRY, KeyVaultKeyNearExpiryEventData.class);
-        put(KEY_VAULT_KEY_EXPIRED, KeyVaultKeyExpiredEventData.class);
-        put(KEY_VAULT_SECRET_NEW_VERSION_CREATED, KeyVaultSecretNewVersionCreatedEventData.class);
-        put(KEY_VAULT_SECRET_NEAR_EXPIRY, KeyVaultSecretNearExpiryEventData.class);
-        put(KEY_VAULT_SECRET_EXPIRED, KeyVaultSecretExpiredEventData.class);
-        put(KEY_VAULT_VAULT_ACCESS_POLICY_CHANGED, KeyVaultAccessPolicyChangedEventData.class);
-    }};
+    private static final Map<String, Class<?>> SYSTEM_EVENT_MAPPINGS = new HashMap<String, Class<?>>() {
```

Comment:
(not for this release) - Keeping the generated system events in sync with this map manually can become hard to maintain. As a follow-up action, we should consider using the customization tool to generate this map along with the rest of the code.

Reply:
Agree. We should automate this. Very time consuming to maintain.

Reply:
Feature crew is following up on this after GA